electrode_models/nucleation_single_particle/initialize.py

```diff
@@ -0,0 +1,129 @@
+"""
+    single_particle_electrode.py
+
+    Class file for single-particle (SPM) electrode methods
+"""
+
+import cantera as ct
+import numpy as np
+
+def initialize(input_file, inputs, sep_inputs, counter_inputs, electrode_name, 
+        params, offset):
+    """
+    Initialize the model.
+    """ 
+    class electrode:
+        """
+        Create an electrode object representing the dense electrode
+        """
+
+        # Import relevant Cantera objects.
+        bulk_obj = ct.Solution(input_file, inputs['bulk-phase'])
+        elyte_obj = ct.Solution(input_file, inputs['electrolyte-phase'])
+        conductor_obj = ct.Solution(input_file, inputs['conductor-phase'])
+        surf_obj = ct.Interface(input_file, inputs['surf-phase'], 
+            [bulk_obj, elyte_obj, conductor_obj])
+
+        # Anode or cathode? Positive external current delivers positive charge 
+        # to the anode, and removes positive charge from the cathode.
+        name = electrode_name
+        if name=='anode':
+            i_ext_flag = -1
+        elif name=='cathode':
+            i_ext_flag = 1
+        else:
+            raise ValueError("Electrode must be an anode or a cathode.")
+
+        # Store the species index of the Li ion in the Cantera object for the 
+        # electrolyte phase:
+        index_Li = elyte_obj.species_index(inputs['mobile-ion'])
+
+        # Electrode thickness and inverse thickness:
+        dy = inputs['thickness']
+        dyInv = 1/dy
+
+        # Phase volume fractions
+        eps_solid = inputs['eps_solid']
+        eps_elyte = 1 - eps_solid
+
+        # Electrode-electrolyte interface area, per unit geometric area.
+        # This calculation assumes spherical particles of a single radius, with 
+        # no overlap.
+        A_surf_ratio = (3*eps_solid*dy/inputs['r_p'])
+
+        # Inverse double layer capacitance, per unit interfacial area.
+        C_dl_Inv = 1/inputs['C_dl']
+
+        # Microstructure-based transport scaling factor, based on Bruggeman 
+        # coefficient of -0.5:
+        elyte_microstructure = eps_elyte**1.5
+        
+        # SV_offset specifies the index of the first SV variable for the 
+        # electrode (zero for anode, nVars_anode + nVars_sep for the cathode)
+        SV_offset = offset
+
+        # Determine the electrode capacity (Ah/m2)
+
+        # Max concentration of stored ion (intercalated Li)
+        # Save initial X
+        X_o = bulk_obj.X 
+        # Set object concentration to fully lithiated:
+        bulk_obj.X = inputs['stored-ion']['name']+':1.0' 
+        # Concentration of stored Li, per unit volume of intercalation phase:
+        C = bulk_obj[inputs['stored-ion']['name']].concentrations[0]
```

Comment:
You will want a different routine.  Basically, what is the maximum capacity of the cathode?

I think it would be based on the amount of Li stored if you could fill all the pores with LiO2, correct?